### ISSUE TITLE
Nativefier 7.7.1 using old Gulp crashes at runtime with Node >= 11

### ISSUE THREAD

**Issue by dukejones** [jiahaog/nativefier#816]
### Description

Hello, I spent much of the day trying to get `gulp build` to run so I could start on a PR... what I'd like to know is, what version of node and npm do you use to develop this?  I tried 3 or 4 versions.  

### Steps to reproduce issue

Show up with the latest node.  `npm run dev-up` and fail.

### Details

- Version of Nativefier (run `nativefier --version`): `master branch`
- Version of node.js (run `node --version`): `v12.1.0`
- OS: `OS X Mojave`
- Error message / stack trace (if any):

```
> nativefier@7.6.12 build /Users/dukejones/work/hack/Native/nativefier
> gulp build

[13:30:13] Requiring external module @babel/register
Trace: The node type SpreadProperty has been renamed to SpreadElement
    at Object.isSpreadProperty (/Users/dukejones/work/hack/node_modules/@babel/types/lib/validators/generated/index.js:4080:11)
    at hasSpread (/Users/dukejones/work/hack/Native/nativefier/node_modules/babel-plugin-transform-object-rest-spread/lib/index.js:38:13)
    at PluginPass.ObjectExpression (/Users/dukejones/work/hack/Native/nativefier/node_modules/babel-plugin-transform-object-rest-spread/lib/index.js:234:14)
    at newFn (/Users/dukejones/work/hack/node_modules/@babel/traverse/lib/visitors.js:237:21)
    at NodePath._call (/Users/dukejones/work/hack/node_modules/@babel/traverse/lib/path/context.js:65:20)
    at NodePath.call (/Users/dukejones/work/hack/node_modules/@babel/traverse/lib/path/context.js:40:17)
    at NodePath.visit (/Users/dukejones/work/hack/node_modules/@babel/traverse/lib/path/context.js:100:12)
    at TraversalContext.visitQueue (/Users/dukejones/work/hack/node_modules/@babel/traverse/lib/context.js:142:16)
    at TraversalContext.visitMultiple (/Users/dukejones/work/hack/node_modules/@babel/traverse/lib/context.js:97:17)
    at TraversalContext.visit (/Users/dukejones/work/hack/node_modules/@babel/traverse/lib/context.js:180:19)
Trace: The node type SpreadProperty has been renamed to SpreadElement
    at Object.isSpreadProperty (/Users/dukejones/work/hack/node_modules/@babel/types/lib/validators/generated/index.js:4080:11)
    at hasSpread (/Users/dukejones/work/hack/Native/nativefier/node_modules/babel-plugin-transform-object-rest-spread/lib/index.js:38:13)
    at PluginPass.ObjectExpression (/Users/dukejones/work/hack/Native/nativefier/node_modules/babel-plugin-transform-object-rest-spread/lib/index.js:234:14)
    at newFn (/Users/dukejones/work/hack/node_modules/@babel/traverse/lib/visitors.js:237:21)
    at NodePath._call (/Users/dukejones/work/hack/node_modules/@babel/traverse/lib/path/context.js:65:20)
    at NodePath.call (/Users/dukejones/work/hack/node_modules/@babel/traverse/lib/path/context.js:40:17)
    at NodePath.visit (/Users/dukejones/work/hack/node_modules/@babel/traverse/lib/path/context.js:100:12)
    at TraversalContext.visitQueue (/Users/dukejones/work/hack/node_modules/@babel/traverse/lib/context.js:142:16)
    at TraversalContext.visitMultiple (/Users/dukejones/work/hack/node_modules/@babel/traverse/lib/context.js:97:17)
    at TraversalContext.visit (/Users/dukejones/work/hack/node_modules/@babel/traverse/lib/context.js:180:19)
Trace: The node type SpreadProperty has been renamed to SpreadElement
    at Object.isSpreadProperty (/Users/dukejones/work/hack/node_modules/@babel/types/lib/validators/generated/index.js:4080:11)
    at hasSpread (/Users/dukejones/work/hack/Native/nativefier/node_modules/babel-plugin-transform-object-rest-spread/lib/index.js:38:13)
    at PluginPass.ObjectExpression (/Users/dukejones/work/hack/Native/nativefier/node_modules/babel-plugin-transform-object-rest-spread/lib/index.js:234:14)
    at newFn (/Users/dukejones/work/hack/node_modules/@babel/traverse/lib/visitors.js:237:21)
    at NodePath._call (/Users/dukejones/work/hack/node_modules/@babel/traverse/lib/path/context.js:65:20)
    at NodePath.call (/Users/dukejones/work/hack/node_modules/@babel/traverse/lib/path/context.js:40:17)
    at NodePath.visit (/Users/dukejones/work/hack/node_modules/@babel/traverse/lib/path/context.js:100:12)
    at TraversalContext.visitQueue (/Users/dukejones/work/hack/node_modules/@babel/traverse/lib/context.js:142:16)
    at TraversalContext.visitSingle (/Users/dukejones/work/hack/node_modules/@babel/traverse/lib/context.js:102:19)
    at TraversalContext.visit (/Users/dukejones/work/hack/node_modules/@babel/traverse/lib/context.js:182:19)
fs.js:27
const { Math, Object, Reflect } = primordials;
                                  ^

ReferenceError: primordials is not defined
    at fs.js:27:35
    at req_ (/Users/dukejones/work/hack/Native/nativefier/node_modules/natives/index.js:143:24)
    at Object.req [as require] (/Users/dukejones/work/hack/Native/nativefier/node_modules/natives/index.js:55:10)
    at Object.<anonymous> (/Users/dukejones/work/hack/Native/nativefier/node_modules/vinyl-fs/node_modules/graceful-fs/fs.js:1:37)
    at Module._compile (internal/modules/cjs/loader.js:759:30)
    at Module._compile (/Users/dukejones/work/hack/node_modules/pirates/lib/index.js:91:24)
    at Module._extensions..js (internal/modules/cjs/loader.js:770:10)
    at Object.newLoader [as .js] (/Users/dukejones/work/hack/node_modules/pirates/lib/index.js:96:7)
    at Module.load (internal/modules/cjs/loader.js:628:32)
    at Function.Module._load (internal/modules/cjs/loader.js:555:12)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! nativefier@7.6.12 build: `gulp build`
npm ERR! Exit status 1
```

**Comment by dukejones** [comment]
I went to the Travis build to see what they were using.

```

$ node --version
v11.14.0
$ npm --version
6.7.0
$ nvm --version
0.34.0
before_install
9.23s$ npm install -g npm@5.8.x
/home/travis/.nvm/versions/node/v11.14.0/bin/npm -> /home/travis/.nvm/versions/node/v11.14.0/lib/node_modules/npm/bin/npm-cli.js
/home/travis/.nvm/versions/node/v11.14.0/bin/npx -> /home/travis/.nvm/versions/node/v11.14.0/lib/node_modules/npm/bin/npx-cli.js
+ npm@5.8.0
added 412 packages from 148 contributors, removed 315 packages and updated 46 packages in 8.722s
install
33.53s$ npm run dev-up
```

And still errors.  Different errors, but errors nonetheless.

```

> nativefier@7.6.12 build /Users/dukejones/work/hack/Native/nativefier
> gulp build

[15:49:07] Requiring external module @babel/register
Trace: The node type SpreadProperty has been renamed to SpreadElement
    at Console.trace (internal/console/constructor.js:336:11)
    at Object.isSpreadProperty (/Users/dukejones/work/hack/node_modules/@babel/types/lib/validators/generated/index.js:4080:11)
    at hasSpread (/Users/dukejones/work/hack/Native/nativefier/node_modules/babel-plugin-transform-object-rest-spread/lib/index.js:38:13)
    at PluginPass.ObjectExpression (/Users/dukejones/work/hack/Native/nativefier/node_modules/babel-plugin-transform-object-rest-spread/lib/index.js:234:14)
    at newFn (/Users/dukejones/work/hack/node_modules/@babel/traverse/lib/visitors.js:237:21)
    at NodePath._call (/Users/dukejones/work/hack/node_modules/@babel/traverse/lib/path/context.js:65:20)
    at NodePath.call (/Users/dukejones/work/hack/node_modules/@babel/traverse/lib/path/context.js:40:17)
    at NodePath.visit (/Users/dukejones/work/hack/node_modules/@babel/traverse/lib/path/context.js:100:12)
    at TraversalContext.visitQueue (/Users/dukejones/work/hack/node_modules/@babel/traverse/lib/context.js:142:16)
    at TraversalContext.visitMultiple (/Users/dukejones/work/hack/node_modules/@babel/traverse/lib/context.js:97:17)
Trace: The node type SpreadProperty has been renamed to SpreadElement
    at Console.trace (internal/console/constructor.js:336:11)
    at Object.isSpreadProperty (/Users/dukejones/work/hack/node_modules/@babel/types/lib/validators/generated/index.js:4080:11)
    at hasSpread (/Users/dukejones/work/hack/Native/nativefier/node_modules/babel-plugin-transform-object-rest-spread/lib/index.js:38:13)
    at PluginPass.ObjectExpression (/Users/dukejones/work/hack/Native/nativefier/node_modules/babel-plugin-transform-object-rest-spread/lib/index.js:234:14)
    at newFn (/Users/dukejones/work/hack/node_modules/@babel/traverse/lib/visitors.js:237:21)
    at NodePath._call (/Users/dukejones/work/hack/node_modules/@babel/traverse/lib/path/context.js:65:20)
    at NodePath.call (/Users/dukejones/work/hack/node_modules/@babel/traverse/lib/path/context.js:40:17)
    at NodePath.visit (/Users/dukejones/work/hack/node_modules/@babel/traverse/lib/path/context.js:100:12)
    at TraversalContext.visitQueue (/Users/dukejones/work/hack/node_modules/@babel/traverse/lib/context.js:142:16)
    at TraversalContext.visitMultiple (/Users/dukejones/work/hack/node_modules/@babel/traverse/lib/context.js:97:17)
Trace: The node type SpreadProperty has been renamed to SpreadElement
    at Console.trace (internal/console/constructor.js:336:11)
    at Object.isSpreadProperty (/Users/dukejones/work/hack/node_modules/@babel/types/lib/validators/generated/index.js:4080:11)
    at hasSpread (/Users/dukejones/work/hack/Native/nativefier/node_modules/babel-plugin-transform-object-rest-spread/lib/index.js:38:13)
    at PluginPass.ObjectExpression (/Users/dukejones/work/hack/Native/nativefier/node_modules/babel-plugin-transform-object-rest-spread/lib/index.js:234:14)
    at newFn (/Users/dukejones/work/hack/node_modules/@babel/traverse/lib/visitors.js:237:21)
    at NodePath._call (/Users/dukejones/work/hack/node_modules/@babel/traverse/lib/path/context.js:65:20)
    at NodePath.call (/Users/dukejones/work/hack/node_modules/@babel/traverse/lib/path/context.js:40:17)
    at NodePath.visit (/Users/dukejones/work/hack/node_modules/@babel/traverse/lib/path/context.js:100:12)
    at TraversalContext.visitQueue (/Users/dukejones/work/hack/node_modules/@babel/traverse/lib/context.js:142:16)
    at TraversalContext.visitSingle (/Users/dukejones/work/hack/node_modules/@babel/traverse/lib/context.js:102:19)
/Users/dukejones/work/hack/node_modules/@babel/traverse/lib/scope/index.js:978
      scope.bindings[name] = info;
            ^

TypeError: Cannot read property 'bindings' of null
    at Scope.moveBindingTo (/Users/dukejones/work/hack/node_modules/@babel/traverse/lib/scope/index.js:978:13)
    at BlockScoping.updateScopeInfo (/Users/dukejones/work/hack/Native/nativefier/node_modules/babel-plugin-transform-es2015-block-scoping/lib/index.js:364:17)
    at BlockScoping.run (/Users/dukejones/work/hack/Native/nativefier/node_modules/babel-plugin-transform-es2015-block-scoping/lib/index.js:330:12)
    at PluginPass.BlockStatementSwitchStatementProgram (/Users/dukejones/work/hack/Native/nativefier/node_modules/babel-plugin-transform-es2015-block-scoping/lib/index.js:70:24)
    at newFn (/Users/dukejones/work/hack/node_modules/@babel/traverse/lib/visitors.js:237:21)
    at NodePath._call (/Users/dukejones/work/hack/node_modules/@babel/traverse/lib/path/context.js:65:20)
    at NodePath.call (/Users/dukejones/work/hack/node_modules/@babel/traverse/lib/path/context.js:40:17)
    at NodePath.visit (/Users/dukejones/work/hack/node_modules/@babel/traverse/lib/path/context.js:100:12)
    at TraversalContext.visitQueue (/Users/dukejones/work/hack/node_modules/@babel/traverse/lib/context.js:142:16)
    at TraversalContext.visitSingle (/Users/dukejones/work/hack/node_modules/@babel/traverse/lib/context.js:102:19)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! nativefier@7.6.12 build: `gulp build`
npm ERR! Exit status 1
```

**Comment by mxdpeep** [comment]
Gulp 3 is no more supported

**Comment by ronjouch** [comment]
Yeah. The current version works for me on Linux & Node 10, but breaks at node 11 / 12. I'm in the middle of a cleanup that will get rid of Gulp and many other outdated dependencies. Until then, it's not ideal. Try node 10.

**Comment by ronjouch** [comment]
Fixed in Nativefier 8.0.2, which drops Gulp. Give it a try 🙂.
